overlord/snapstate/snapstate.go

```diff
@@ -406,6 +426,52 @@ func TryPath(st *state.State, name, path string, flags Flags) (*state.TaskSet, e
 	return InstallPath(st, &snap.SideInfo{RealName: name}, path, "", flags)
 }
 
+// ensureCore ensures that the core snap is installed
+func ensureCore(st *state.State, snapName string, userID int) (*state.TaskSet, error) {
+	ts := &state.TaskSet{}
+
+	if snapName == defaultCoreSnapName || snapName == "ubuntu-core" {
+		return ts, nil
+	}
+
+	// check that there is no task that installs core already
+	for _, chg := range st.Changes() {
+		if chg.Status().Ready() {
+			continue
+		}
+		for _, t := range chg.Tasks() {
+			if t.Kind() == "link-snap" {
+				snapsup, err := TaskSnapSetup(t)
+				if err != nil {
+					return nil, err
+				}
+				// some other change aleady installs core
+				if snapsup.Name() == defaultCoreSnapName {
```

Comment:
this is problematic because we could end up with this snap but without core if that other core install failed, or try to run things too early in this install that might need core to be there. I fear we need to give up instead for now or construct a cross change dependency which we never tried